### PR TITLE
Converted most similarity values from double to float.

### PR DESCRIPTION
**Pull request by allan-shoup**

### CODE REVIEW

File: src/main/java/org/scify/jedai/progressivejoin/SetSimJoin.java [hunk 1]
```diff
@@ -26,10 +27,9 @@
 public class SetSimJoin {
 
     private LinkedHashMap<String, ArrayList<Integer>> records;
-    private ArrayList<Object[]> results;
```
Comment (allan-shoup) [inline src/main/java/org/scify/jedai/progressivejoin/SetSimJoin.java:29]: I was a bit confused by the format of these object arrays so I thought I'd change it to a first class object. Originally I created
```
public class PairSimilairty (
    float similarity;
    int recordId1;
    int recordId2;
}
```
But then noticed it was pretty similar to the existing `Comparison` object so I opted to use that instead. If you find the usage of `Comparison` unclear, I'd be fine adding back a class like `PairSimilarity`.

File: src/main/java/org/scify/jedai/progressivejoin/TopkGlobal.java [hunk 2]
```diff
@@ -54,12 +57,11 @@ else if (Math.abs((Double) o1[2]-(Double) o2[2])<1E-13 && (int)o1[0] == (int)o2[
         });
         records.forEach((recordid, tokens) -> events.add(new Object[] {recordid, 0, 1.0}));
 
-        // <simval, recordid1, recordid2>
-        SortedList<Object[]> tmpresults = new SortedList<>((o1, o2) -> {
-            if ((Double)o1[0] > (Double)o2[0])
```
Comment (allan-shoup) [inline src/main/java/org/scify/jedai/progressivejoin/TopkGlobal.java:59]: Please double-check the logic change here. I moved the `return 0` case in front of the `return -1` thinking the intent was that similarity values that were very close (within some rounding error) were to be considered equal.

**Comment by gpapadis** [comment]
Thank you very much indeed, Allan!

**Comment by allan-shoup** [comment]
You're welcome, and thank you for the quick response!

**Pull request merged**